src/Package.php

```diff
@@ -623,6 +639,27 @@ public function statusIs(int $status): bool
         return $this->checkStatus($status);
     }
 
+    /**
+     * @return bool
+     */
+    public function hasFailed(): bool
+    {
+        return $this->status === self::STATUS_FAILED;
+    }
+
+    /**
+     * @param int $status
+     * @return bool
+     */
+    public function hasReachedStatus(int $status): bool
+    {
+        if (!isset(self::SUCCESS_STATUSES[$status])) {
+            return false;
+        }
+
+        return !$this->hasFailed() && $this->checkStatus($status, '>=');
```

Comment:
I know, this is just restructuring code, but I think this can be a little bit easier to read, and we actually get rid of two `!` operators (which is not the main point, of course):

```suggestion
        if ($this->hasFailed()) {
            return false;
        }

        return isset(self::SUCCESS_STATUSES[$status]) && $this->checkStatus($status, '>=');
```

Reply:
Normally I follow the rule of thumb: less expensive checks first. But I'm fine sacrificing that rule in this case for better readabiility.